src/backend/tests/routers/test_agent.py

```diff
@@ -736,22 +736,24 @@ def test_update_agent_with_tool_metadata_and_new_tool_metadata(
     )
 
     assert response.status_code == 200
-    updated_agent = response.json()
 
     tool_metadata = (
         session.query(AgentToolMetadata)
         .filter(AgentToolMetadata.agent_id == agent.id)
         .all()
     )
     assert len(tool_metadata) == 2
-    assert tool_metadata[0].tool_name == "google_drive"
```

Comment:
there is no guarantee that `tool_metadata[0]` is always the `google_drive` tool, this leads to flaky behaviour